### PR TITLE
Amend provisioned concurrency examples and templates to work for false condition

### PR DESCRIPTION
**Pull request by me2resh**
*Issue #, if available:*
None

*Description of changes:*
The current examples and templates for lambda provisioned concurrency works fine for a true valid condition, but if the condition in the example is false, it will fail with and error because the parent property ProvisionedConcurrencyConfig will still be present, but will have no child configuration settings.

*Description of how you validated changes:*
In the example templates, change the condition AliasProvisionedConcurrencyEnabled to false
Then when you try to deploy the template, cloudformation deploy will fail with the error:
```
Property ProvisionedConcurrentExecutions cannot be empty.
```

I also added a missing condition in the lambda_edge/template.yaml 

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [x] Update documentation
- [x] Verify transformed template deploys and application functions as expected
- [x] Add/update example to `examples/2016-10-31`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by praneetap** [comment]
Changing base to develop.

File: examples/2016-10-31/cloudwatch-event-to-msteams/template.yaml [hunk 1]
```diff
@@ -45,6 +45,7 @@ Resources:
           Type: CloudWatchEvent
           Description: Detects EC2 Security Group Events to Send to Teams
           Properties:
+            EventBusName:  event-bus-name
```
Comment (praneetap) [inline examples/2016-10-31/cloudwatch-event-to-msteams/template.yaml]: could you undo this change?

**Comment by me2resh** [comment]
Hi @praneetap, the branch currently is rebased against the latest develop and has only my intended changes. please re-review.
